source/adapters/level_zero/context.hpp

```diff
@@ -222,45 +217,42 @@ struct ur_context_handle_t_ : _ur_object {
   };
 
   std::list<ze_event_pool_handle_t> *
-  getZeEventPoolCache(bool HostVisible, bool WithProfiling,
-                      bool CounterBasedEventEnabled, bool UsingImmediateCmdList,
-                      ze_device_handle_t ZeDevice) {
+  getZeEventPoolCache(EventFlags Flags, ze_device_handle_t ZeDevice) {
     EventPoolCacheType CacheType;
 
-    calculateCacheIndex(HostVisible, CounterBasedEventEnabled,
-                        UsingImmediateCmdList, CacheType);
+    calculateCacheIndex(Flags, CacheType);
     if (ZeDevice) {
       auto ZeEventPoolCacheMap =
-          WithProfiling ? &ZeEventPoolCacheDeviceMap[CacheType * 2]
-                        : &ZeEventPoolCacheDeviceMap[CacheType * 2 + 1];
+          Flags & ENABLE_PROFILER
+              ? &ZeEventPoolCacheDeviceMap[CacheType * 2]
+              : &ZeEventPoolCacheDeviceMap[CacheType * 2 + 1];
       if (ZeEventPoolCacheMap->find(ZeDevice) == ZeEventPoolCacheMap->end()) {
         ZeEventPoolCache.emplace_back();
         ZeEventPoolCacheMap->insert(
             std::make_pair(ZeDevice, ZeEventPoolCache.size() - 1));
       }
       return &ZeEventPoolCache[(*ZeEventPoolCacheMap)[ZeDevice]];
     } else {
-      return WithProfiling ? &ZeEventPoolCache[CacheType * 2]
-                           : &ZeEventPoolCache[CacheType * 2 + 1];
+      return Flags & ENABLE_PROFILER ? &ZeEventPoolCache[CacheType * 2]
+                                     : &ZeEventPoolCache[CacheType * 2 + 1];
     }
   }
 
-  ur_result_t calculateCacheIndex(bool HostVisible,
-                                  bool CounterBasedEventEnabled,
-                                  bool UsingImmediateCmdList,
+  ur_result_t calculateCacheIndex(EventFlags Flags,
                                   EventPoolCacheType &CacheType) {
-    if (CounterBasedEventEnabled && HostVisible && !UsingImmediateCmdList) {
```

Comment:
hm, can't we just use Flags as the index?

Reply:
As in using the flags for the index of zeEventPoolCache?

Reply:
Yes. Notice that the number of combinations in your cache grows exponentially with the number of boolean flags (+ potentially number of bits you'd need to dedicate to store the "id" of a device). So a flag, with the corresponding bits set, is your index into the cache.